src/pages/case-studies/[slug].astro

```diff
@@ -0,0 +1,175 @@
+---
+import MainLayout from "~/layouts/MainLayout.astro"
+import { CollectionEntry, getCollection, getEntryBySlug } from "astro:content"
+import { format } from "date-fns"
+import Avatar from "~/components/Avatar.astro"
+import {
+	resolveBlogImage,
+	resolveCoverImage,
+	resolveSocialImage,
+} from "~/content/blog/_resolveImage.js"
+import { randomFromArray } from "~/helpers/random.js"
+import Layout from "~/layouts/MainLayout.astro"
+import houstonWink from "~/assets/houston_wink.webp"
+
+export const prerender = true
+
+export async function getStaticPaths() {
+	const items = await getCollection("caseStudies")
+	return items.map((post) => ({
+		params: { slug: post.slug },
+		props: { post },
+	}))
+}
+
+type Props = {
+	post: CollectionEntry<"caseStudies">
+}
+
+const { post } = Astro.props
+
+const { data, render } = post
+const { Content } = await render()
+
+// const coverImage = await resolveCoverImage(post)
+// const socialImage = await resolveSocialImage(post)
+
+const headerImage = await resolveBlogImage(post.data.headerImage)
+console.log(headerImage)
+
+const authors = await Promise.all(
+	data.authors.map((author) => getEntryBySlug("authors", author)),
+).then((authors) => authors.filter(Boolean) as CollectionEntry<"authors">[])
+
+const gradients = [
+	"bg-red-pink-gradient",
+	"bg-blue-green-gradient",
+	"bg-blue-purple-gradient",
+	"bg-orange-yellow-gradient",
+]
+
+const placeholderGradient = randomFromArray(gradients)
+---
+
+<!-- <Layout
```

Comment:
make sure to pull these comments out before merging, Astro will keep all of them in the DOM 😅